ui/src/config.js

```diff
@@ -63,16 +61,34 @@ export const appConfig = {
   },
 };
 
-export const sentryConfig = {
+const sentryConfig = {
   dsn: process.env.REACT_APP_SENTRY_DSN,
   environment: appConfig.environment,
+  tags: {
+    app: "turing-ui",
+  },
 };
 
-export const resultLoggingConfig = {
+const resultLoggingConfig = {
   protoUrl: process.env.REACT_APP_RESULT_LOG_PROTO_URL,
 };
 
-export const alertConfig = {
+const alertConfig = {
   environment:
     appConfig.environment === "dev" ? "development" : appConfig.environment,
 };
+
+const buildTimeConfig = {
+  apiConfig,
+  authConfig,
+  appConfig,
+  sentryConfig,
+  resultLoggingConfig,
+  alertConfig,
+};
+
+export const useConfig = () => {
+  const runTimeConfig = window.config;
```

Comment:
What is this sorcery? Where are we saving the dynamic config from `app.config.js`  to the window?

Reply:
`config` is a global variable defined in `app.config.js` and any global variables can be accessed as properties of `window` object. ([See More](https://developer.mozilla.org/en-US/docs/Glossary/Global_object#window_object_in_the_browser))